nodes/server.js

```diff
@@ -75,104 +75,38 @@ module.exports = function(RED) {
             node.devices_values = {};
         }
 
-        getDevices(callback, forceRefresh = false, withGroups = false) {
-            var node = this;
-
-            if (forceRefresh || node.devices === undefined) {
-                node.log('Refreshing devices');
-                node.groups = [];
-
-                var timeout = null;
-                var timeout_ms = 60000;
-
-                var client = node.connectMQTT('tmp');
-                client.on('connect', function() {
-
-                    //end function after timeout, if now response
+        getDevices(callback, withGroups = false) {
+            if (typeof (callback) !== 'function') {
+                return
+            }
+            let node = this;
+            if (node.devices && (!withGroups || node.groups)) {
+                node.log('Using cached devices')
+                callback(withGroups ? [node.devices, node.groups] : node.devices);
+            } else {
+                node.log('Waiting for device list')
+                let timeout = null
+                let checkAvailability = null
+                new Promise(function(resolve) {
                     timeout = setTimeout(function() {
-                        node.error('Error: getDevices timeout, close connection')
-                        client.end(true);
-                    }, timeout_ms);
-
-                    client.subscribe(node.getTopic('/#'), {'qos':parseInt(node.config.mqtt_qos||0)}, function(err) {
-                        if (err) {
-                            node.error('Error code #0023: ' + err);
-                            client.end(true);
+                        resolve()
+                    }, 60_000);
```

Comment:
Typo in setTimeout function?
Value should be 60000 instead of 60_000.

Reply:
https://v8.dev/features/numeric-separators

Reply:
Strange.
I got a syntax error on line 93.
Just removing the underscore in my local source file fixed the issue.

I'm running Node-Red in a docker container (nodered/node-red)
NodeJS version in this container is v16.20.2